include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1033,33 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Compute a vector of bounding boxes which encloses the space occupied by
+   * the cells which have the predicate property
+   *
+   * The algorithm loops over all cells at given level @p refinement_level and active cells at lower refinement
+   * levels, creating a Bounding Box for each of them. Then it merges "redundant" boxes i.e. those which can
+   * be expressed as a single one. In a last optional step all smaller boxes are merged with neighbours.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose. See this function's test for an example.
+   *  The predicate is tested only on active cells.
+   * - @p refinement_level : it describes the level at which the initial Bounding Boxes are created. The refinement
+   *  should be set to a low refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation an exception is thrown.
+   * - @p max_boxes : the maximum number of bounding boxes to compute. If more are created the smaller ones are
+   *  merged with neighbours. By default all boxes are kept after merging and erasing the reduntant ones.
+   *  Notice only neighbouring cells are merged (see the @p is_neighbour function in bounding box class): if
+   *  the target number of bounding boxes max_boxes can't be reached by merging neighbours an exception is thrown
+   *
+   * Notice this function doesn't work with curved meshes.
```

Comment:
@note This function does not take into account curvature of cells, i.e. it's assumed that the mapping is linear.